components/eamxx/tpls/CMakeLists.txt

```diff
@@ -16,3 +16,61 @@ if (SCREAM_CIME_BUILD)
   include (${SCREAM_BASE_DIR}/cmake/tpls/Mct.cmake)
   CreateMctTarget()
 endif()
+
+# MAM aerosol support
+if (SCREAM_ENABLE_MAM)
```

Comment:
I'm curious, why did you use ExternalProject_Add rather than just to add_subdirectory for haero/mam4xx, like we do for other submodules?

Reply:
I prefer `ExternalProject_Add` to in-tree builds because it's more of a self-contained system that plays more nicely with other things. You can build a software package and have it install its things in a given location without having to understand what variables it sets or any of that business, and you can just look at the installed config CMake file (or pkgconfig or whatever) to fish out the information.

By contrast, I find in-tree builds harder to understand, because they are sucked inside their parent projects, and it's not as clear what is dictating the behavior of the compilation. Also, if you have packages that assume that another software package is already installed in some prefix, it's harder to convince them that an in-tree software package is actually installed as it would be "in the field." I've had this experience with Sandia's TChem library, which wants a pre-installed version of Kokkos somewhere, and the way we build EKAT (and Kokkos) in-tree creates a lot of complexity.

Happy to discuss, adjust, etc. But I hope you agree with me that installing Haero and mam4xx this way is pretty simple!

Reply:
Good points. Relying on an installation of some of our TPLs would make CIME-vs-standalone similar. I might revisit my stance on subdirs...